pandas/tests/io/test_sql.py

```diff
@@ -1305,6 +1621,10 @@ def test_api_roundtrip(conn, request, test_frame1):
     result = sql.read_sql_query("SELECT * FROM test_frame_roundtrip", con=conn)
 
     # HACK!
+    if "adbc" in conn_name:
```

Comment:
I guess this is just what happens when you use `Table.from_pandas` on the pyarrow side. Not sure if we care that much to handle this in the `to_sql` call, leave the test as is. @jorisvandenbossche may also know of some pyarrow tricks

Reply:
In theory in `to_sql` we could give the index a name if it has none. Looking at how this is handled for the existing engines, it seems we have a `_index_name` helper function that currently lives on `SQLTable`, and that could be moved to a standalone helper function so we can use it there and also in the new adbc to_sql 

The different name indeed comes from `pyarrow.Table.from_pandas` using different names as what pandas uses here.